aicssegmentation/workflow/workflow_definition.py

```diff
@@ -20,24 +20,31 @@ class WorkflowDefinition:
 
     name: str
     steps: List[WorkflowStep]
+    from_file: bool
 
     @lazy_property
     def thumbnail_pre(self) -> np.ndarray:
         """
         The Pre-segmentation thumbnail related to this workflow, as a numpy array
         """
+        if self.from_file:
+            raise FileNotFoundError("Cannot get thumbnails for custom loaded workflows.")
```

Comment:
This is kind of an anti-pattern - we normally don't want to throw exceptions for valid actions. In this case if these properties are exposed it should be perfectly valid to call them without expecting an error.

Here are two alternative proposed alternatives:

- (this isn't great but it's better) return null or an empty ndarray 
- (preferred solution) use inheritance since we essentially have 2 "flavors" of workflow definitions, one with more functionality than the other. The base `WorkflowDefinition` can just have the name and steps but no images. And derive a child class for example `StandardWorkflowDefinition` (not in love with the name, open to other suggestions) that adds the thumbnail and diagram properties. You can then change the `get_executable_workflow` to return this new class, and the `get_executable_workflow_from_file` that you create can return the regular `WorklowDefinition`.